stan/math/rev/fun/gp_exp_quad_cov.hpp

```diff
@@ -208,82 +27,83 @@ class gp_exp_quad_cov_vari<T_x, double, T_l> : public vari {
  * @throw std::domain_error if sigma <= 0, l <= 0, or
  *   x is nan or infinite
  */
-template <typename T_x,
-          typename = require_arithmetic_t<typename scalar_type<T_x>::type>>
-inline Eigen::Matrix<var, -1, -1> gp_exp_quad_cov(const std::vector<T_x> &x,
-                                                  const var &sigma,
-                                                  const var &length_scale) {
-  check_positive("gp_exp_quad_cov", "sigma", sigma);
-  check_positive("gp_exp_quad_cov", "length_scale", length_scale);
-  size_t x_size = x.size();
-  for (size_t i = 0; i < x_size; ++i) {
+template <typename T_x, typename T_sigma, typename T_l,
+          require_stan_scalar_t<T_sigma>* = nullptr,
+          require_stan_scalar_t<T_l>* = nullptr,
+          require_any_st_var<T_x, T_sigma, T_l>* = nullptr>
+inline Eigen::Matrix<var, -1, -1> gp_exp_quad_cov(const std::vector<T_x>& x,
+                                                  T_sigma sigma,
+                                                  T_l length_scale) {
+  check_positive("gp_exp_quad_cov", "marginal standard deviation", sigma);
+  check_positive("gp_exp_quad_cov", "length scale", length_scale);
+  for (size_t i = 0; i < x.size(); ++i) {
     check_not_nan("gp_exp_quad_cov", "x", x[i]);
   }
```

Comment:
```suggestion
  check_not_nan("gp_exp_quad_cov", "x", x);
```

Reply:
This pull got reverted. It's waiting over here to get back in: https://github.com/stan-dev/math/pull/2007